pytorch_lightning/trainer/distrib_parts.py

```diff
@@ -498,7 +499,7 @@ def single_gpu_train(self, model):
 
     def tpu_train(self, tpu_core_idx, model):
         # put model on tpu
-        model.to(xm.xla_device())
+        model.to(xm.xla_device(self.tpu_id))
```

Comment:
i think this now makes it ONLY possible to train on 1 core no? not multiple cores

Reply:
I think so... @lezwon ^^

Reply:
I have noticed that if `self.tpu_id` is `None` and I use `xmp.spawn`, the model trains at the same speed it trains when all cores are being used. So I assumed that all cores are being used. I could add some logging to confirm. Or just add a conditional for `xm.xla_device()` maybe?

Reply:
ONLY when the user requests a specific TPU index should we use 
```model.to(xm.xla_device(self.tpu_id))``` otherwise, leave it as it was.

@Borda we need TPU tests to make sure this PR doesn't break functionality